src/visualization/components/internal/StaticLegend.tsx

```diff
@@ -91,14 +92,15 @@ const StaticLegend: FC<Props> = ({properties, update}) => {
   const heightRatioRef = useRef<number>(null)
   heightRatioRef.current = heightRatio
 
-  React.useEffect(() => {
-    if (
-      heightRatio === STATIC_LEGEND_HEIGHT_RATIO_NOT_SET &&
-      heightRatio !== defaultHeightRatio
-    ) {
-      setHeightRatio(defaultHeightRatio)
+  useEffect(() => {
+    setHeightRatio(defaultHeightRatio)
+  }, [defaultHeightRatio])
+
+  useEffect(() => {
+    if (showOptions && heightRatio === STATIC_LEGEND_HEIGHT_RATIO_NOT_SET) {
+      updateHeightRatio(STATIC_LEGEND_HEIGHT_RATIO_DEFAULT)
     }
-  }, [defaultHeightRatio, heightRatio])
+  }, [showOptions])
```

Comment:
If open the "show" on the legend, and it has the NOT_SET being used --> then override with the default hieght.